hypervisor/debug/profiling.c

```diff
@@ -112,19 +112,26 @@ static void profiling_enable_pmu(void)
 		__func__, MSR_IA32_EXT_APIC_LVT_PMI, lvt_perf_ctr);
 
 	if (ss->guest_debugctl_value != 0U) {
-		/* Set the VM Exit MSR Load in VMCS */
+		/* Merge the msr vmexit loading list with HV */
 		if (ss->vmexit_msr_cnt == 0) {
-			ss->vmexit_msr_cnt = 1;
-			ss->vmexit_msr_list[0].msr_idx
+			struct acrn_vcpu *vcpu = get_ever_run_vcpu(get_cpu_id());
+
+			ss->vmexit_msr_cnt = 1 + MSR_AREA_COUNT;
+			ss->vmexit_msr_list[0].msr_num
 				= MSR_IA32_DEBUGCTL;
-			ss->vmexit_msr_list[0].msr_data
+			ss->vmexit_msr_list[0].value
 				= ss->guest_debugctl_value &
 					VALID_DEBUGCTL_BIT_MASK;
 
+			for (i = 0; i < MSR_AREA_COUNT; i++) {
+				ss->vmexit_msr_list[i + 1].msr_num = vcpu->arch.msr_area.host[i].msr_num;
+				ss->vmexit_msr_list[i + 1].value = vcpu->arch.msr_area.host[i].value;
+			}
+
 			exec_vmwrite64(VMX_EXIT_MSR_LOAD_ADDR_FULL,
 				hva2hpa(ss->vmexit_msr_list));
-			exec_vmwrite(VMX_EXIT_MSR_LOAD_COUNT,
-				(uint64_t)ss->vmexit_msr_cnt);
+			exec_vmwrite32(VMX_EXIT_MSR_LOAD_COUNT,
+				(uint32_t)ss->vmexit_msr_cnt);
```

Comment:
we could convert vmexit_msr_cnt to uint32_t.  Up to you.

Reply:
merge into one line

Reply:
Fixed